source/JsMaterialX/JsMaterialXCore/JsGeom.cpp

```diff
@@ -1,58 +1,72 @@
+#include "../vectorHelper.h"
 #include "../helpers.h"
+
 #include <MaterialXCore/Geom.h>
 
-#include <emscripten.h>
 #include <emscripten/bind.h>
 
 namespace ems = emscripten;
 namespace mx = MaterialX;
 
+using stRef = const std::string&;
+
 #define BIND_GEOMINFO_FUNC_INSTANCE(NAME, T) \
-.function("_setGeomPropValue" #NAME, &mx::GeomInfo::setGeomPropValue<T>)
+    BIND_MEMBER_FUNC("setGeomPropValue" #NAME, mx::GeomInfo, setGeomPropValue<T>, 2, 3, stRef, const T&, stRef)
 
 extern "C"
 {
     EMSCRIPTEN_BINDINGS(geom)
     {
+        ems::constant("GEOM_PATH_SEPARATOR", mx::GEOM_PATH_SEPARATOR);
+        ems::constant("UNIVERSAL_GEOM_NAME", mx::UNIVERSAL_GEOM_NAME);
+        ems::constant("UDIM_TOKEN", mx::UDIM_TOKEN);
+        ems::constant("UDIMSET", mx::UDIMSET);
+        ems::constant("UV_TILE_TOKEN", mx::UV_TILE_TOKEN);       
+
         ems::class_<mx::GeomElement, ems::base<mx::Element>>("GeomElement")
             .smart_ptr<std::shared_ptr<mx::GeomElement>>("GeomElement")
             .smart_ptr<std::shared_ptr<const mx::GeomElement>>("GeomElement")
             .function("setGeom", &mx::GeomElement::setGeom)
             .function("hasGeom", &mx::GeomElement::hasGeom)
             .function("getGeom", &mx::GeomElement::getGeom)
+            .function("getActiveGeom", &mx::GeomElement::getActiveGeom)
             .function("setCollectionString", &mx::GeomElement::setCollectionString)
             .function("hasCollectionString", &mx::GeomElement::hasCollectionString)
             .function("getCollectionString", &mx::GeomElement::getCollectionString)
             .function("setCollection", &mx::GeomElement::setCollection)
-            .function("getCollection", &mx::GeomElement::getCollection);
+            .function("getCollection", &mx::GeomElement::getCollection)
+            // BIND_VALIDATE(mx::GeomElement)
```

Comment:
Why is this commented?

Reply:
Enabling it will break the tests whenever `doc.validated()` is invoked for some not yet discovered reason. Good catch it was something we wanted to look at again.